parser/src/main/scala/aqua/parser/lexer/LambdaOp.scala

```diff
@@ -39,16 +40,18 @@ object LambdaOp {
   private val parseField: P[LambdaOp[Span.S]] =
     (`.` *> `name`).lift.map(IntoField(_))
 
-  private val nonNegativeIntP0: P0[Int] =
-    Numbers.nonNegativeIntString.map(_.toInt).?.map(_.getOrElse(0))
-
   private val parseIdx: P[LambdaOp[Span.S]] =
-    P.defer(
+    (P.defer(
       (ValueToken.`value`.between(`[`, `]`) | (exclamation *> ValueToken.num))
         .map(v => IntoIndex(v, Some(v)))
         .backtrack
-    ) |
-      exclamation.lift.map(e => IntoIndex(Token.lift[Span.S, Unit](e), None))
+    ) | exclamation.lift.map(e => IntoIndex(Token.lift[Span.S, Unit](e), None))).flatMap { ii =>
+      ii.idx match {
+        case Some(LiteralToken(_, lt)) if lt == LiteralType.signed =>
+          P.fail.withContext("Indexes in the array must be positive numbers")
```

Comment:
`Collection indexes must be non-negative` or smth like that? It could be not an array